ibis/sql/alchemy.py

```diff
@@ -82,7 +86,32 @@
 _sqla_type_to_ibis.update(_sqla_type_mapping)
 
 
-def sqlalchemy_type_to_ibis_type(column_type, nullable=True):
+_numpy_dtype_map = {
+    # SQLAlchemyType: (non-nullable dtype, nullable dtype)
+    sa.SmallInteger: ('int16', 'float64'),
+    sa.SMALLINT: ('int16', 'float64'),
+    sa.Integer: ('int32', 'float64'),
+    sa.INTEGER: ('int32', 'float64'),
+    sa.BigInteger: ('int64', 'float64'),
+    sa.BIGINT: ('int64', 'float64'),
+    sa.Boolean: ('bool', 'object'),
+    sa.BOOLEAN: ('bool', 'object'),
```

Comment:
It does tie to pandas missing value semantics which isn't great. Let me see if this is still necessary.